Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp

```diff
@@ -1106,6 +1137,12 @@ void GraphicsContextCG::didUpdateState(GraphicsContextState& state)
             setCGShadow(renderingMode(), state.dropShadow().offset, state.dropShadow().blurRadius, state.dropShadow().color, state.shadowsIgnoreTransforms());
             break;
 
+        case GraphicsContextState::Change::Style:
+#if HAVE(CORE_GRAPHICS_STYLES)
+            setCGStyle(context, state.style());
+#endif
+            break;
```

Comment:
Should we be asserting in here in the non-`HAVE(CORE_GRAPHICS_STYLES)` state, or will we be unconditionally calling `GraphicsContextState::setStyle`, but skipping the existing accelerated / software implementations in the `HAVE(CORE_GRAPHICS_STYLES)` case?

Reply:
`GraphicsContext::setStyle()` is not necessarily used by filters. For example we could implement `GraphicsContext::setDropShadow()` in terms of `GraphicsContext::setStyle()`.